File: src/tools/stats.ts
// stats — small aggregate over the cached snapshot.

import { statusClass } from "../format/http-parse.js";
import { renderStats } from "../format/render.js";
import type { SnapshotStore } from "../snapshot.js";

export async function stats(store: SnapshotStore): Promise<string> {
  const snap = await store.get();
  const byMethod: Record<string, number> = {};
  const byClass: Record<string, number> = {};
  const byHost: Record<string, number> = {};

  for (const e of snap.entries) {
    const m = e.request.method || "-";
    byMethod[m] = (byMethod[m] ?? 0) + 1;
    const cls = statusClass(e.response.status);
    byClass[cls] = (byClass[cls] ?? 0) + 1;
    if (e.request.host) {
      byHost[e.request.host] = (byHost[e.request.host] ?? 0) + 1;
    }
  }

  const topHosts = Object.entries(byHost)
    .sort((a, b) => b[1] - a[1])
    .slice(0, 5)
    .map(([h, n]) => [h, n] as const);

  return renderStats({
    total: snap.count,
    byMethod,
    byClass,
    byHost: topHosts,
  });
}

File: src/burp_mcp_lite/tools/stats.py
"""stats — small aggregate over the cached snapshot."""

from __future__ import annotations

from collections import Counter

from ..format.http_parse import status_class
from ..format.render import render_stats
from ..snapshot import SnapshotStore


async def stats(store: SnapshotStore) -> str:
    snap = await store.get()
    by_method: Counter[str] = Counter()
    by_class: Counter[str] = Counter()
    by_host: Counter[str] = Counter()
    for e in snap.entries:
        by_method[e.request.method or "-"] += 1
        by_class[status_class(e.response.status)] += 1
        if e.request.host:
            by_host[e.request.host] += 1
    top_hosts = by_host.most_common(5)
    return render_stats(
        total=snap.count,
        by_method=dict(by_method),
        by_class=dict(by_class),
        by_host=top_hosts,
    )
